0.9.2: internalnotification uses kdb_long_double_t without checking

Hi,

this problem has already been addressed once in 8743cdc06585f4d867b936337f92568a830ddbe1 but apparently got reintroduced.

## Steps to Reproduce the Problem
Try compiling elektra for an armel based platform.

## Expected Result
Elektra compiles successfully.

## Actual Result
Compilation fails with this error message:
```
In file included from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:54,
                 from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:458:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:454:14: error: unknown type name 'kdb_long_double_t'; did you mean 'kdb_double_t'?
 #define TYPE kdb_long_double_t
              ^~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/include/macros/type_create_to_value.h:101:58: note: in definition of macro 'TYPE_CONVERSION_SIGNATURE'
  int NAME_MACRO (TYPE_NAME) (const Key * KEY_PARAM_NAME, TYPE * VARIABLE_PARAM_NAME)
                                                          ^~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/include/macros/type_create_to_value.h:122:28: note: in expansion of macro 'TYPE'
 TYPE_CONVERSION_SIGNATURE (TYPE, TYPE_NAME, NAME_MACRO)
                            ^~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h: In function 'elektraInternalnotificationConvertCallbackKdbLongDouble':
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:454:14: error: unknown type name 'kdb_long_double_t'; did you mean 'kdb_double_t'?
 #define TYPE kdb_long_double_t
              ^~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:59:2: note: in expansion of macro 'TYPE'
  TYPE * variable = (TYPE *) ctx->variable;
  ^~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:454:14: error: 'kdb_long_double_t' undeclared (first use in this function); did you mean 'kdb_double_t'?
 #define TYPE kdb_long_double_t
              ^~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:59:21: note: in expansion of macro 'TYPE'
  TYPE * variable = (TYPE *) ctx->variable;
                     ^~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:454:14: note: each undeclared identifier is reported only once for each function it appears in
 #define TYPE kdb_long_double_t
              ^~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:59:21: note: in expansion of macro 'TYPE'
  TYPE * variable = (TYPE *) ctx->variable;
                     ^~~~
In file included from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:458:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:59:27: error: expected expression before ')' token
  TYPE * variable = (TYPE *) ctx->variable;
                           ^
In file included from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/include/kdbplugin.h:15,
                 from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/include/kdbnotificationinternal.h:18,
                 from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.h:14,
                 from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:10:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:43:82: warning: implicit declaration of function 'elektraInternalnotificationConvertKdbLongDouble'; did you mean 'elektraInternalnotificationConvertKdbDouble'? [-Wimplicit-function-declaration]
 #define INTERNALNOTIFICATION_CONVERSION_FUNCTION_NAME(TYPE_NAME) ELEKTRA_CONCAT (elektraInternalnotificationConvert, TYPE_NAME)
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/include/kdbmacros.h:21:31: note: in definition of macro 'ELEKTRA_CONCAT2'
 #define ELEKTRA_CONCAT2(X, Y) X##Y
                               ^
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:43:66: note: in expansion of macro 'ELEKTRA_CONCAT'
 #define INTERNALNOTIFICATION_CONVERSION_FUNCTION_NAME(TYPE_NAME) ELEKTRA_CONCAT (elektraInternalnotificationConvert, TYPE_NAME)
                                                                  ^~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:60:7: note: in expansion of macro 'INTERNALNOTIFICATION_CONVERSION_FUNCTION_NAME'
  if (!INTERNALNOTIFICATION_CONVERSION_FUNCTION_NAME (TYPE_NAME) (key, variable))
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:458:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h: At top level:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:454:14: error: unknown type name 'kdb_long_double_t'; did you mean 'kdb_double_t'?
 #define TYPE kdb_long_double_t
              ^~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:47:82: note: in definition of macro 'INTERNALNOTIFICATION_REGISTER_SIGNATURE'
  int INTERNALNOTIFICATION_REGISTER_NAME (TYPE_NAME) (Plugin * handle, Key * key, TYPE * variable)
                                                                                  ^~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/macros/add_type.h:69:42: note: in expansion of macro 'TYPE'
 INTERNALNOTIFICATION_REGISTER_SIGNATURE (TYPE, TYPE_NAME)
                                          ^~~~
In file included from /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:10:
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c: In function 'elektraInternalnotificationGet':
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.h:28:55: error: 'elektraInternalnotificationRegisterKdbLongDouble' undeclared (first use in this function); did you mean 'elektraInternalnotificationRegisterKdbDouble'?
 #define INTERNALNOTIFICATION_REGISTER_NAME(TYPE_NAME) elektraInternalnotificationRegister##TYPE_NAME
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.h:28:55: note: in definition of macro 'INTERNALNOTIFICATION_REGISTER_NAME'
 #define INTERNALNOTIFICATION_REGISTER_NAME(TYPE_NAME) elektraInternalnotificationRegister##TYPE_NAME
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/src/plugins/internalnotification/internalnotification.c:540:54: note: in expansion of macro 'INTERNALNOTIFICATION_EXPORT_FUNCTION'
    INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbDouble), INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbLongDouble),
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[5]: *** [src/plugins/internalnotification/CMakeFiles/elektra-internalnotification-objects.dir/build.make:80: src/plugins/internalnotification/CMakeFiles/elektra-internalnotification-objects.dir/internalnotification.c.o] Error 1
make[5]: Leaving directory '/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/build'
make[4]: *** [CMakeFiles/Makefile2:5197: src/plugins/internalnotification/CMakeFiles/elektra-internalnotification-objects.dir/all] Error 2
make[4]: Leaving directory '/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/build'
make[3]: *** [Makefile:180: all] Error 2
make[3]: Leaving directory '/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/build'
make[2]: *** [Makefile:450: /home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_arm926ej-s_musl_eabi/elektra-0.9.2/.built] Error 2
make[2]: Leaving directory '/home/harald/packages/libs/elektra'
time: package/feeds/custom/elektra/compile#41.79#9.32#51.70
make[1]: *** [package/Makefile:113: package/feeds/custom/elektra/compile] Error 2
make[1]: Leaving directory '/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64'
make: *** [/home/harald/openwrt-sdk-mxs_gcc-8.4.0_musl_eabi.Linux-x86_64/include/toplevel.mk:217: package/elektra/compile] Fehler 2
```

## System Information

- Elektra Version: 0.9.2
- OpenWRT SDK for mxs

## Further Log Files and Output

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -81,6 +81,9 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 - The `cache` plugin now only caches the parts of the global keyset that are below `system:/elektra/cache` or below
   `system:/elektra/cached`. The part below `system:/elektra/cache` is meant for internal data of the `cache`, so you
   should put data below `system:/elektra/cached`, if you want it to be cached. _(Klemens Böswirth)_
+### internalnotification
+
+- Fix use of `kdb_long_double_t` on armel platforms ([#3450](https://github.com/ElektraInitiative/libelektra/issues/3450)). _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -81,6 +81,7 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 - The `cache` plugin now only caches the parts of the global keyset that are below `system:/elektra/cache` or below
   `system:/elektra/cached`. The part below `system:/elektra/cache` is meant for internal data of the `cache`, so you
   should put data below `system:/elektra/cached`, if you want it to be cached. _(Klemens Böswirth)_
+
 ### internalnotification
 
 - Fix use of `kdb_long_double_t` on armel platforms ([#3450](https://github.com/ElektraInitiative/libelektra/issues/3450)). _(Mihael Pranjić)_

--- a/src/libs/notification/notification.c
+++ b/src/libs/notification/notification.c
@@ -89,7 +89,9 @@ ELEKTRA_NOTIFICATION_TYPE_DEFINITION (kdb_long_long_t, KdbLongLong)
 ELEKTRA_NOTIFICATION_TYPE_DEFINITION (kdb_unsigned_long_long_t, KdbUnsignedLongLong)
 ELEKTRA_NOTIFICATION_TYPE_DEFINITION (kdb_float_t, KdbFloat)
 ELEKTRA_NOTIFICATION_TYPE_DEFINITION (kdb_double_t, KdbDouble)
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
 ELEKTRA_NOTIFICATION_TYPE_DEFINITION (kdb_long_double_t, KdbLongDouble)
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
 
 int elektraNotificationRegisterCallback (KDB * kdb, Key * key, ElektraNotificationChangeCallback callback, void * context)
 {

--- a/src/plugins/internalnotification/internalnotification.c
+++ b/src/plugins/internalnotification/internalnotification.c
@@ -455,11 +455,13 @@ void elektraInternalnotificationUpdateRegisteredKeys (Plugin * plugin, KeySet * 
 #define CHECK_CONVERSION ELEKTRA_TYPE_CHECK_CONVERSION
 #include "macros/add_type.h"
 
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
 #define TYPE kdb_long_double_t
 #define TYPE_NAME KdbLongDouble
 #define TO_VALUE (strtold (string, &end))
 #define CHECK_CONVERSION ELEKTRA_TYPE_CHECK_CONVERSION
 #include "macros/add_type.h"
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
 
 /**
  * @see kdbnotificationinternal.h ::ElektraNotificationPluginRegisterCallback
@@ -538,7 +540,10 @@ int elektraInternalnotificationGet (Plugin * handle, KeySet * returned, Key * pa
 			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbUnsignedShort), INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbLong),
 			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbUnsignedLong), INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbLongLong),
 			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbUnsignedLongLong), INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbFloat),
-			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbDouble), INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbLongDouble),
+			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbDouble),
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
+			INTERNALNOTIFICATION_EXPORT_FUNCTION (KdbLongDouble),
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
 
 			keyNew ("system:/elektra/modules/internalnotification/exports/registerCallback", KEY_FUNC,
 				elektraInternalnotificationRegisterCallback, KEY_END),

--- a/src/plugins/internalnotification/testmod_internalnotification.c
+++ b/src/plugins/internalnotification/testmod_internalnotification.c
@@ -706,6 +706,7 @@ static void test_doUpdateShouldNotUpdateUnregisteredKey (void)
 #define CHECK_INVALID ((int) value == 0)
 #include "macros/create_type_tests.h"
 
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
 #define TYPE kdb_long_double_t
 #define TYPE_NAME KdbLongDouble
 #define FORMAT_STRING "%1.8f"
@@ -714,6 +715,7 @@ static void test_doUpdateShouldNotUpdateUnregisteredKey (void)
 #define INVALID_VALUE "4.a"
 #define CHECK_INVALID ((int) value == 0)
 #include "macros/create_type_tests.h"
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
 
 int main (int argc, char ** argv)
 {
@@ -763,7 +765,9 @@ int main (int argc, char ** argv)
 	RUN_TYPE_TESTS (KdbUnsignedLongLong)
 	RUN_TYPE_TESTS (KdbFloat)
 	RUN_TYPE_TESTS (KdbDouble)
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
 	RUN_TYPE_TESTS (KdbLongDouble)
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
 
 	printf ("\nelektraInternalnotificationDoUpdate\n-----------------------------------\n");
 	test_doUpdateShouldUpdateKey ();

